tinygrad/codegen/optimizer.py

```diff
@@ -349,7 +345,8 @@ def hand_coded_optimizations(self):
     # upcast leading axes first (hack-ish for winograd; we actually want to upcast masked axes with low stride first)
     for axis in range(self.first_reduce):
       # we might want to be able to split axes that are masked, or refuse to merge them in simplify_merge_adjacent
-      if self.full_shape[axis] <= 7 and any(st.axis_is_masked(axis) for st in self.sts) and prod(self.full_shape[self.shape_len - self.upcasted:]) * self.full_shape[axis] <= 7 * 7:
+      # for now skip upcasting here if there is a symbolic axis
+      if isinstance(self.full_shape[axis], int) and self.full_shape[axis] <= 7 and any(st.axis_is_masked(axis) for st in self.sts) and prod(self.full_shape[self.shape_len - self.upcasted:]) * self.full_shape[axis] <= 7 * 7:
```

Comment:
self.full_shape[axis] <= 7 was the problem here.
The symbolic shapes has a weird __bool__ definition where every Variable is True when the var.max or var.min > 1.
This causes this to be true for example:
bool(Variable(None, 2,17) <= 7) == 1

I think this should be changed but this takes more effort